src/dda/utils/fs.py

```diff
@@ -152,6 +152,30 @@ def __as_exe(self) -> Path:
         def __as_exe(self) -> Path:
             return self
 
+    @classmethod
+    def enc_hook(cls, obj: Any) -> Any:
```

Comment:
I would prefer that we not add these and modify the caller like here https://github.com/DataDog/datadog-agent-dev/blob/ec05716465b2e7d8dea425ecdb25067d9ae5a32f/src/dda/config/model/__init__.py#L57-L70

Maybe we can have a more general mechanism but I haven't thought much about that.